jwt-auth-common/src/main/java/io/telicent/smart/caches/configuration/auth/RemoteUserInfoLookup.java

```diff
@@ -79,11 +79,18 @@ public UserInfo lookup(String userInfoEndpoint, String bearerToken) throws UserI
             } else {
                 throw new UserInfoLookupException("Unexpected status " + status + " from userinfo endpoint. Body: " + body);
             }
-        } catch (IOException | InterruptedException ex) {
+        } catch (InterruptedException ex) {
             Thread.currentThread().interrupt();
+            throw new UserInfoLookupException("Request interrupted", ex);
+        } catch (IOException ex) {
             throw new UserInfoLookupException("I/O error when calling userinfo endpoint", ex);
         } catch (IllegalArgumentException ex) {
             throw new UserInfoLookupException("Invalid userInfoEndpoint URL", ex);
         }
     }
+
+    @Override
+    public void close() throws IOException {
+        // HttpClient does not allow for explicit shutdown
```

Comment:
You're quite right, but so was I, apparently the ability to shutdown a client was only added in JDK 21 (https://stackoverflow.com/a/76563407/107591) and as these libraries currently remain JDK 17 compatible we can't do that yet